### PR TITLE
DOC: add missing parameters to offsets classes: BQuarterBegin, BQuarterEnd, QuarterBegin, QuarterEnd

### PR DESCRIPTION
**Pull request by natmokval**
xref  #52431. 
Updated documentation for offsets classes. Added missing parameters and See Also section to `BQuarterBegin`, `BQuarterEnd`, `QuarterBegin`, `QuarterEnd`.

### CODE REVIEW

File: pandas/_libs/tslibs/offsets.pyx [hunk 1]
```diff
@@ -2506,6 +2506,19 @@ cdef class BQuarterEnd(QuarterOffset):
     startingMonth = 2 corresponds to dates like 2/28/2007, 5/31/2007, ...
     startingMonth = 3 corresponds to dates like 3/30/2007, 6/29/2007, ...
 
+    Parameters
+    ----------
+    n : int, default 1
+        The number of years represented.
```
Comment (MarcoGorelli) [inline pandas/_libs/tslibs/offsets.pyx:2512]: is this years or quarters?

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
thanks - generally looks good, just got a question about `n`

File: pandas/_libs/tslibs/offsets.pyx (hunk 1 above)
Comment (natmokval) [reply pandas/_libs/tslibs/offsets.pyx:2512]: thanks, @MarcoGorelli , you are right.  n represents quarters, my mistake. I''l correct this.

**Review by MarcoGorelli — APPROVED** [review]
thanks @natmokval !

**Pull request merged**